src/main/java/bookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package bookstore.service.impl;
+
+import bookstore.dto.category.CategoryDto;
+import bookstore.exception.EntityNotFoundException;
+import bookstore.mapper.CategoryMapper;
+import bookstore.model.Category;
+import bookstore.repository.CategoryRepository;
+import bookstore.service.CategoryService;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.mapstruct.Named;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@Named("CategoryService")
```

Comment:
```suggestion
```

Reply:
changed algorithm of categories initialization while Book create/update requests